### PR TITLE
chore: Extract `pages project list` tests into a separate file

### PR DESCRIPTION
**Pull request by CarmenPopoviciu**
All Pages-related tests currently live under one file, `/__tests__/pages.test.ts`. Over time, this file has grown in size and complexity, and our tests have become very hard to follow or read.

This commit works towards solving this problem by separating our tests into separate, dedicated files 🧡 

What this PR solves / how to test:

Associated docs issues/PR:

- [insert associated docs issue(s)/PR(s)]

Author has included the following, where applicable:

- [ ] Tests
- [ ] Changeset

Reviewer has performed the following, where applicable:

- [ ] Checked for inclusion of relevant tests
- [ ] Checked for inclusion of a relevant changeset
- [ ] Checked for creation of associated docs updates
- [ ] Manually pulled down the changes and spot-tested

Fixes # [insert issue number].

### CODE REVIEW

File: packages/wrangler/src/__tests__/helpers/mock-get-pages-upload-token.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { rest } from "msw";
+import { msw } from "./msw";
+
+export function mockGetUploadToken(
```
Comment (jahands) [inline packages/wrangler/src/__tests__/helpers/mock-get-pages-upload-token.ts:4]: Mind adding a docstring for what this does?

File: packages/wrangler/src/__tests__/helpers/end-event-loop.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export function endEventLoop() {
```
Comment (jahands) [inline packages/wrangler/src/__tests__/helpers/end-event-loop.ts:1]: This is probably simple but I'm not super familiar with event loops so a docstring explaining what it does (and maybe why you'd use it) would be helpful

**Review by GregBrimble — APPROVED** [review]
If this is just 99% a copy-paste job, approved! If you did change anything of substance, please flag it :)

File: packages/wrangler/src/__tests__/pages.test.ts [hunk 3]
```diff
@@ -1,56 +1,23 @@
-/* eslint-disable no-shadow */
```
Comment (jahands) [inline packages/wrangler/src/__tests__/pages.test.ts:1]: nit: Any reason we shouldn't put all Pages tests in the `pages` directory? If I'm looking for Pages tests, I might not notice that I need to check two places. I noticed D1 also has all of it's test files in the `d1` directory, so it'd be nice to be consistent with that as well.

File: packages/wrangler/src/__tests__/pages/deployment-list.test.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import type { Deployment } from "./../../pages/types";
+
+describe("deployment list", () => {
+	runInTempDir();
+	mockAccountId();
+	mockApiToken();
+
+	beforeEach(() => {
+		// @ts-expect-error we're using a very simple setTimeout mock here
+		jest.spyOn(global, "setTimeout").mockImplementation((fn, _period) => {
```
Comment (jahands) [inline packages/wrangler/src/__tests__/pages/deployment-list.test.ts:16]: Would it be worth breaking this out into a shared helper function so we don't have to copy/paste to every file?

File: packages/wrangler/src/__tests__/pages/publish.test.ts [hunk 5]
```diff
@@ -0,0 +1,2168 @@
+/* eslint-disable no-shadow */
```
Comment (jahands) [inline packages/wrangler/src/__tests__/pages/publish.test.ts:1]: Is there any way it would make sense to break these tests down into separate files? (No worries if not)

**Review by jahands — COMMENTED** [review]
This is awesome! Happy to see these tests become more manageable

File: packages/wrangler/src/__tests__/helpers/mock-get-pages-upload-token.ts (hunk 1 above)
Comment (CarmenPopoviciu) [reply packages/wrangler/src/__tests__/helpers/mock-get-pages-upload-token.ts:4]: ✅

File: packages/wrangler/src/__tests__/pages.test.ts (hunk 3 above)
Comment (CarmenPopoviciu) [reply packages/wrangler/src/__tests__/pages.test.ts:1]: ✅

File: packages/wrangler/src/__tests__/helpers/end-event-loop.ts (hunk 2 above)
Comment (CarmenPopoviciu) [reply packages/wrangler/src/__tests__/helpers/end-event-loop.ts:1]: ✅

File: packages/wrangler/src/__tests__/pages/publish.test.ts (hunk 5 above)
Comment (CarmenPopoviciu) [reply packages/wrangler/src/__tests__/pages/publish.test.ts:1]: these are all `publish` related tests. I don't have a good idea yet on how we can better organise these. did you have smth in mind?

File: packages/wrangler/src/__tests__/pages/deployment-list.test.ts (hunk 4 above)
Comment (CarmenPopoviciu) [reply packages/wrangler/src/__tests__/pages/deployment-list.test.ts:16]: ✅

**Review by GregBrimble — APPROVED** [review]
Yeah, LGTM!

**Review by jahands — APPROVED** [review]
LGTM!

File: packages/wrangler/src/__tests__/pages/publish.test.ts (hunk 5 above)
Comment (jahands) [reply packages/wrangler/src/__tests__/pages/publish.test.ts:1]: No I didn't... This is great as-is :)
